src/app/posts/state/post.effects.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from "@angular/core";
+import { Actions, createEffect, ofType } from "@ngrx/effects";
+import { of } from "rxjs";
+import { catchError, map, mergeMap } from "rxjs/operators";
+import { PostService } from "../post.service";
+import { PostActions } from "./actions";
+
+@Injectable()
+export class PostEffects {
+    constructor(
+        private actions$: Actions,
+        private postService: PostService
+    ) { }
+
+    loadBlogPosts$ = createEffect(() => {
+        return this.actions$.pipe(
+            ofType(PostActions.loadPosts),
+            mergeMap(() => this.postService.getPosts().pipe(
```

Comment:
You may use mergeMap if you need to handle multiple requests concurrently.
Here however switchMap would be more appropriate. It will cancel previous Observable in case another action is dispatched.

See more info here - [https://blog.angular-university.io/rxjs-higher-order-mapping/](https://blog.angular-university.io/rxjs-higher-order-mapping/)